internals/search_queries.py

```diff
@@ -74,32 +74,36 @@ def total_order_query_async(sort_spec):
   return keys_promise
 
 
-def sorted_by_pending_request_date(descending):
-  """Return feature_ids of pending approvals sorted by request date."""
-  query = models.Approval.query(
-      models.Approval.state == models.Approval.REVIEW_REQUESTED)
+def _sorted_by_joined_model(
+    joinable_model_class, condition, descending, order_by):
+  """Return feature_ids sorted by a field in a joined entity kind."""
+  query = joinable_model_class.query()
+  if condition:
+    query = query.filter(condition)
   if descending:
-    query = query.order(-models.Approval.set_on)
+    query = query.order(-order_by)
   else:
-    query = query.order(models.Approval.set_on)
+    query = query.order(order_by)
 
-  pending_approvals = query.fetch(projection=['feature_id'])
-  feature_ids = utils.dedupe(pa.feature_id for pa in pending_approvals)
+  joined_models = query.fetch(projection=['feature_id'])
+  feature_ids = utils.dedupe(jm.feature_id for jm in joined_models)
   return feature_ids
 
 
+def sorted_by_pending_request_date(descending):
+  """Return feature_ids of pending approvals sorted by request date."""
+  return _sorted_by_joined_model(
+      models.Approval,
+      models.Approval.state == models.Approval.REVIEW_REQUESTED,
```

Comment:
`models.Approval.state == models.Approval.REVIEW_REQUESTED` would this part be evaluated automatically to True or. False by Python?

Reply:
ndb is tricky:  It's not a boolean at all, it evaluates to a `ndb.FilterNode` object.   `models.Approval.state` is an `ndb.IntegerProperty`.  Its  base class `ndb.Property` overrides `__eq__` to return a `ndb.FilterNode` object rather than a boolean.   We make use of that overloaded operator all the time in calls to SomeModelClass.query() and some_query.filter().

Reply:
` overrides __eq__ to return a ndb.FilterNode` -> aha interesting choice here